### PR TITLE
Move sawtooth cluster new cli

### PR DESCRIPTION
**Pull request by ojalatodd**

### CODE REVIEW

**Comment by ojalatodd** [comment]
@vaporos  hmmm... I rebased but the tests failed. Seems to be related to genesis stuff. 
@peterschwarz  - might have input?

**Comment by ojalatodd** [comment]
Moved to external: https://github.com/hyperledger/

**Comment by ojalatodd** [comment]
The test that is failing is also failing for master, according to @rbuysse . 
Note: all tests pass locally on the vagrant VM:

1. run_lint
2. run_tests

File: bin/sawtooth [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
-# Copyright 2016 Intel Corporation
+# Copyright 2017 Intel Corporation
```
Comment (boydjohnson) [inline bin/sawtooth]: According to @TomBarnes, the correct way to update a copyright is "Copyright 2016, 2017 Intel Corporation" if the file was created in 2016 and updated in 2017.

Comment (ojalatodd) [inline bin/sawtooth]: Fixed

**Comment by ojalatodd** [comment]
All tests pass locally, but on the build server ( build.sawtooth.me ), test fail here:
======================================================================
ERROR: sawtooth_suites.ts_pr_dev_mode.DevModeTestSuite (nose2.loader.LoadTestsFailure)
----------------------------------------------------------------------
AttributeError: 'module' object has no attribute 'ts_pr_dev_mode'

**Comment by ojalatodd** [comment]
@boydjohnson and @vaporos 
Tests are passing locally, but there's a hitch on the server as described above. Trying to get help from Ryan or Adam.
The test's error is very non-useful to me in tracking this down. 
Would like to get this PR merged soon.

File: cli/sawtooth_cli/main.py [hunk 2]
```diff
@@ -95,11 +95,8 @@ def create_parser(prog_name):
     add_admin_parser(subparsers, parent_parser)
     add_config_parser(subparsers, parent_parser)
     add_block_parser(subparsers, parent_parser)
-<<<<<<< 8a234e6efe39c7de9acfd8f30978e795f0d7e38c
```
Comment (peterschwarz) [inline cli/sawtooth_cli/main.py]: This fix should be squashed into the earlier commit that caused the conflict.

File: cli/sawtooth_cli/main.py [hunk 3]
```diff
@@ -118,8 +124,13 @@ def main(prog_name=os.path.basename(sys.argv[0]), args=sys.argv[1:],
         do_config(args)
     elif args.command == 'block':
         do_block(args)
+<<<<<<< 8a234e6efe39c7de9acfd8f30978e795f0d7e38c
```
Comment (peterschwarz) [inline cli/sawtooth_cli/main.py]: This needs to be addressed as well.

File: cli/sawtooth_cli/main.py (hunk 2 above)
Comment (ojalatodd) [inline cli/sawtooth_cli/main.py]: fixed

File: cli/sawtooth_cli/main.py (hunk 3 above)
Comment (ojalatodd) [inline cli/sawtooth_cli/main.py]: fixed

**Comment by ojalatodd** [comment]
Tests passed!

File: manage/sawtooth_manage/docker.py [hunk 4]
```diff
@@ -315,3 +315,8 @@ def is_running(self, node_name):
             if node_name == entry.name:
                 return entry.status.startswith("Up")
         return False
+
+
+class CliException(Exception):
```
Comment (peterschwarz) [inline manage/sawtooth_manage/docker.py]: This should be named something specific to manage, since this is not in the CLI subproject

Comment (ojalatodd) [inline manage/sawtooth_manage/docker.py]: Addred

File: integration/sawtooth_integration/docker/intkey-smoke.yaml [hunk 5]
```diff
@@ -47,8 +47,8 @@ services:
       - 40000
     # start the validator with an empty genesis batch
     entrypoint: "bash -c \"\
-        ./bin/sawtooth-0.8 admin genesis && \
-        ./bin/validator -vv\""
+        ./bin/sawtooth admin genesis && \
+        ./bin/validator -v\""
```
Comment (vaporos) [inline integration/sawtooth_integration/docker/intkey-smoke.yaml]: Let's keep the -vv here.

File: integration/sawtooth_integration/docker/xo-smoke.yaml [hunk 6]
```diff
@@ -46,8 +46,8 @@ services:
     expose:
       - 40000
     entrypoint: "bash -c \"\
-        ./bin/sawtooth-0.8 admin genesis && \
-        ./bin/validator -vv\""
+        ./bin/sawtooth admin genesis && \
+        ./bin/validator -v\""
```
Comment (vaporos) [inline integration/sawtooth_integration/docker/xo-smoke.yaml]: Let's keep the -vv here.

File: manage/sawtooth_manage/docker.py [hunk 7]
```diff
@@ -162,25 +161,25 @@ def start(self, node_config):
...
 
         except OSError as e:
             if e.errno == 2:
-                raise CliException("{}:{}".format(str(e), args[0]))
+                raise ManagementError("{}:{}".format(str(e), args[0]))
             else:
                 raise e
 
-        for line in output.split('\n'):
+        for line in output.decode('utf8').split('\n'):
```
Comment (vaporos) [inline manage/sawtooth_manage/docker.py]: utf-8 is the default in python 3 and can be omitted from decode/encode calls.

File: manage/sawtooth_manage/docker.py [hunk 8]
```diff
@@ -216,7 +216,7 @@ def _built_in_processor_types(self):
 
     def stop(self, node_name):
         state_file_path = os.path.join(self._state_dir, 'state.yaml')
-        state = yaml.load(file(state_file_path))
+        state = yaml.load(open(state_file_path))
```
Comment (vaporos) [inline manage/sawtooth_manage/docker.py]: looks like a potential filehandle leak in handling of open here; open should almost always be done:

```
with open(...) as fd:
     s = yaml.load(fd)
```

**Comment by ojalatodd** [comment]
Addressing Shawn's review.

File: integration/sawtooth_integration/docker/intkey-smoke.yaml (hunk 5 above)
Comment (ojalatodd) [inline integration/sawtooth_integration/docker/intkey-smoke.yaml]: Added additional 'v'

File: integration/sawtooth_integration/docker/xo-smoke.yaml (hunk 6 above)
Comment (ojalatodd) [inline integration/sawtooth_integration/docker/xo-smoke.yaml]: Added 'v'

File: manage/sawtooth_manage/docker.py (hunk 7 above)
Comment (ojalatodd) [inline manage/sawtooth_manage/docker.py]: done

File: manage/sawtooth_manage/docker.py (hunk 8 above)
Comment (ojalatodd) [inline manage/sawtooth_manage/docker.py]: Addressed.

File: cli/sawtooth_cli/main.py [hunk 9]
```diff
@@ -93,7 +95,11 @@ def create_parser(prog_name):
     add_admin_parser(subparsers, parent_parser)
     add_config_parser(subparsers, parent_parser)
     add_block_parser(subparsers, parent_parser)
+<<<<<<< 8a234e6efe39c7de9acfd8f30978e795f0d7e38c
```
Comment (ojalatodd) [inline cli/sawtooth_cli/main.py]: yes, but I resolved this. Should I do anything more?

Comment (peterschwarz) [inline cli/sawtooth_cli/main.py]: I think the intent of the change request is to fix that particular commit, not just the code.

**Pull request merged**